addons/purchase/models/purchase.py

```diff
@@ -884,9 +884,8 @@ def _suggest_quantity(self):
         '''
         if not self.product_id:
             return
-
         seller_min_qty = self.product_id.seller_ids\
-            .filtered(lambda r: r.name == self.order_id.partner_id)\
+            .filtered(lambda r: r.name == self.order_id.partner_id and r.product_id == self.product_id)\
```

Comment:
I think you need add
(r.product_id == self.product_id or not r.product_id) instead of r.product_id == self.product_id
For purchase price lists without product variant

Reply:
Thanks!